hikari/impl/shard.py

```diff
@@ -221,16 +228,18 @@ async def _receive_and_check_complete_zlib_package(
     ) -> str:
         buff = bytearray(initial_data)
 
-        while True:
+        while not buff.endswith(_ZLIB_SUFFIX):
```

Comment:
Since we have the fast path above, is this really needed anymore?

Reply:
> since _receive_and_check_complete_zlib_package is a rare case it was left with the change of using while not like that to make it just more resilient and less likely to be quietly broken by changes to the code calling it (it also just looks nicer that way)

(looks nicer meaning it just is nicer code)
(and changes to the code calling it includes the removal of a fast path)
(also this code is rarely called and even under cases where discord split up large payloads it'd still be rarely called so its shouldn't really a performance concern)